risc0/zkvm/sdk/rust/guest/src/env.rs

```diff
@@ -95,8 +95,8 @@ pub(crate) fn finalize(result: *mut usize) {
     ENV.get().finalize(result);
 }
 
-/// Exchanges data with the host, returning the data from the host
-/// as a slice of bytes.
+/// Exchanges data with the host, returning data from the host
+/// as a slice of bytes. In most cases, using env::write() is preferred.
```

Comment:
Maybe word this like:
"See `env::write()` for passing structured data to the host".

Basically it's not really preferred, it just depends on what the needs of the user are. I have a feeling we're going to pretty significantly revamp this part of the API once the zero-copy deserialize lands. There's already some open PRs to provide a better API to allow raw-byte access (which I think will become obsolete when zero-copy lands).